doc/user_guide/psy_data.rst

```diff
@@ -224,30 +226,71 @@ An executable example for using the GOcean read-only-verification
 library is included in ``examples/gocean/eg5/readonly``, see
 :ref:`gocean_example_readonly`.
 
-.. _psydata_nan_test:
+.. _psydata_value_range_check:
 
-NAN Test
---------
+Value Range Check
+-----------------
 
 This transformation can be used for both LFRic and GOcean APIs. It will
-test all input and output parameters of a kernel to make sure they are not
-``NaN`` or infinite. If they are, an error message like the following
-is printed, but the program is not aborted::
+test all input and output parameters of a kernel to make sure they are 
+within a user-specified range. Additionally, it will also very that floating
+point values are not ``NaN`` or infinite. 
+
+At runtime, environment variables must be specified to indicate which variables
+are within what expected range, and optionally also at which location.
+The range is specified as a ``:`` separated tuple::
+
+    1.1:3.3   A value between 1.1 and 3.3 (inclusive).
+    :3.3      A value less than or equal to 3.3
+    1.1:      A value greater than or equal to 1.1
+
+The syntax for the environment variable is one of:
+
+``PSYVERIFY__module__kernel__variable``
```

Comment:
I would think that in basically all cases the form with module name (without kernel) can be used. That should be easy enough. Otherwise you could look in the psy file. I would suggest to leave this for now, and see if we can get user feedback.

Reply:
Agreed.